db/migrate/20250702075600_add_source_metadata_to_captain_documents.rb

```diff
@@ -0,0 +1,11 @@
+class AddSourceMetadataToCaptainDocuments < ActiveRecord::Migration[7.1]
+  def change
+    add_column :captain_documents, :source_type, :string, default: 'url'
+    add_column :captain_documents, :content_type, :string
+    add_column :captain_documents, :file_size, :integer
```

Comment:
The content_type & file_size are the property of the file. It shouldn't be part of the document table. 

Please see how active storage can be used. Here is the guide. https://guides.rubyonrails.org/active_storage_overview.html#has-one-attached